.github/workflows/build.yml

```diff
@@ -0,0 +1,46 @@
+name: build
+on:
+  push:
+    branches:
+      - '*'
+  pull_request: {}
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v3
+        with:
+          go-version-file: 'go.mod'
+      - run: "go vet ./..."
+      - name: Lint
+        uses: dominikh/staticcheck-action@v1.2.0
+        with:
+          version: "2022.1.1"
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v3
```

Comment:
`uses: actions/setup-go@v4`

Reply:
I've be setting to the one used in go mod - we could use a matrix if inclined